hack/devtools/rp_dev_helper.sh

```diff
@@ -0,0 +1,448 @@
+check_vmss() {
+    # Check if exactly non-empty two arguments are provided
+    if [[ $# -ne 2 ]]; then
+        echo "Usage $0 <ResourceGroup> <Name>. Please try again"
+        exit 1
+    fi
+
+    # Check if the VMSS exists
+    vmss_info=$(az vmss show --resource-group $1 --name $2 2>/dev/null)
+    if [ -z "${vmss_info}" ]; then
+        echo "VMSS '$2' in Resource group '$1' does not exist."
+        return 1
+    fi
+
+    # Check if jq is installed
+    if ! check_jq_installed; then
+        exit 1
+    fi
+
+    # Extract the provisioning state using jq
+    provisioning_state=$(echo "${vmss_info}" | jq -r '.properties.provisioningState')
+    # Check if the provisioning state is 'Succeeded'
+    if [[ "${provisioning_state}" == "Succeeded" ]]; then
+        echo "VMSS '$2' in Resource group '$1' has provisioned successfully."
+        return 0
+    else
+        echo "VMSS '$2' in Resource group '$1' has not provisioned successfully. Current state: ${provisioning_state}."
+        return 1
+    fi
+}
+
+
+# Function to cleanup all the created resources from the full aro RP dev
+clean_rp_dev_env() {
+    echo "########## Deleting Dev Env in $LOCATION ##########"
+
+    # Validate number of arguments
+    if [ $# -gt 2 ]; then
+        echo "Usage: $0 [LIST_RESOURCE_GROUPS] [LIST_KEYVAULTS]. Please try again"
+        exit 1
+    fi
+    if [[ $# -ne 2 ]]; then
+        echo "Info: Two input arguments were required. Checking two env vars for default values"
+        # Check if AZURE_PREFIX environment variable are set
+        if [[ -z "${AZURE_PREFIX}" ]]; then
+            echo "Error: AZURE_PREFIX environment variable is not set."
+            exit 1
+        fi
+        echo "Info: AZURE_PREFIX=${AZURE_PREFIX}"
+        # Check if LOCATION environment variable are set
+        if [[ -z "$LOCATION" ]]; then
+            echo "Error: LOCATION environment variable is not set."
+            exit 1
+        fi
+        echo "Info: LOCATION=$LOCATION"
+    fi
+    
+    # Convert input strings to arrays
+    eval "rgs=($1)"
+    eval "kvs=($2)"
+
+   if [[ ${#rgs[@]} -eq 0 ]]; then
+        rg_suffixes=("global" "subscription" "gwy-$LOCATION" "aro-$LOCATION")
+        rgs=()
+        for suffix in "${rg_suffixes[@]}"; do
+            rgs+=("${AZURE_PREFIX}-$suffix")
+        done
+        echo "No resource groups were provided. Use default values for list: ${rgs[*]}"
+    fi
+
+     for rg in "${rgs[@]}"; do
+        echo "########## Delete Resource Group $rg in $LOCATION ##########"
+        az group delete --resource-group "$rg" -y
+    done
+
+    if [[ ${#kvs[@]} -eq 0 ]]; then
+        kv_suffixes=("gwy" "por" "svc" "cls")
+        kvs=()
+        for suffix in "${kv_suffixes[@]}"; do
+            kvs+=("${AZURE_PREFIX}-aro-$LOCATION-$suffix")
+        done
+        echo "No KeyVaults were provided. Use default values for list: ${kvs[*]}"
+    fi
+
+    for kv in "${kvs[@]}"; do
```

Comment:
Don't double quote `kvs` here, it can cause issues with the for loop running, since it's not glob expanded.